### PR TITLE
Enhance gpg case to encrypt file and support fips

### PR DESCRIPTION
**Pull request by dehai**
this case can be running at fips environment and none fips environment.
and also can be changed the generated key size
Verification running at local : http://147.2.212.89/tests/32

### CODE REVIEW

**Comment by dehai** [comment]
related test case : https://bugzilla.suse.com/tr_show_case.cgi?case_id=1595215

File: tests/console/gpg.pm [hunk 1]
```diff
@@ -32,33 +41,91 @@ sub run {
...
+        assert_screen("key-generation-failed");
+    }
+    else {
+        # list gpg keys
+        validate_script_output("gpg --list-keys", sub { m/\[ultimate\] $user_name \<$email\>/ });
+    }
+}
+
+sub gpg_encrypt_file {
+
```
Comment (mitiao) [inline tests/console/gpg.pm]: Delete empty line please

File: tests/console/gpg.pm [hunk 2]
```diff
@@ -32,33 +41,91 @@ sub run {
...
+}
+
+sub run {
+    select_console 'user-console';
+
+    # gpg key generated and file encrypted with two size of key
+    gpg_generate_key(2048);
+    gpg_encrypt_file(2048);
+    gpg_generate_key(3072);
+    gpg_encrypt_file(3072);
```
Comment (mitiao) [inline tests/console/gpg.pm]: How about using loop:
`for (2048, 3072) {
    gpg_generate_key($_);
    gpg_encrypt_file($_);
}`

File: tests/console/gpg.pm (hunk 1 above)
Comment (dehai) [inline tests/console/gpg.pm]: which empty line should be deleted? is it only under sub or all the empty line which include above of annotations - "#"

File: tests/console/gpg.pm (hunk 2 above)
Comment (dehai) [inline tests/console/gpg.pm]: that's ok to change , I will modify it.

**Comment by dehai** [comment]
related needle PR: https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/444

File: tests/console/gpg.pm (hunk 2 above)
Comment (mitiao) [inline tests/console/gpg.pm]: From your failed verified run, I see that implicitly use $_ may cause mistake, so directly assign a  the $key_size variable in loop :)

**Comment by mitiao** [comment]
@Jiawei517 please take care above needle PR, thanks.

**Pull request merged**